src/devices/bus/pce/pce_scdsys.cpp

```diff
@@ -0,0 +1,112 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli, Wilbert Pol
+/***********************************************************************************************************
+
+
+ PC-Engine & Turbografx-16 Super System Card emulation
+
+
+ ***********************************************************************************************************/
+
+
+#include "emu.h"
+#include "pce_scdsys.h"
+
+
+
+//-------------------------------------------------
+//  pce_cdsys3_device - constructor
+//-------------------------------------------------
+
+DEFINE_DEVICE_TYPE(PCE_ROM_CDSYS3_BASE, pce_cdsys3_base_device, "pce_cdsys3_base", "PCE/TG16 Super System Card Base interface")
+DEFINE_DEVICE_TYPE(PCE_ROM_CDSYS3J,     pce_cdsys3j_device,     "pce_cdsys3j",     "PCE Super System Card")
+DEFINE_DEVICE_TYPE(PCE_ROM_CDSYS3U,     pce_cdsys3u_device,     "pce_cdsys3u",     "TG16 Super System Card")
+
+
+pce_cdsys3_base_device::pce_cdsys3_base_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, PCE_ROM_CDSYS3_BASE, tag, owner, clock)
+	, m_region(false)
+{
+}
+
+pce_cdsys3_device::pce_cdsys3_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock, bool region)
+	: device_t(mconfig, type, tag, owner, clock)
+	, device_pce_cart_interface(mconfig, *this)
+	, m_cdsys3(*this, "cdsys3")
+{
+}
+
+pce_cdsys3j_device::pce_cdsys3j_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: pce_cdsys3_device(mconfig, PCE_ROM_CDSYS3J, tag, owner, clock, false)
+{
+}
+
+pce_cdsys3u_device::pce_cdsys3u_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: pce_cdsys3_device(mconfig, PCE_ROM_CDSYS3U, tag, owner, clock, true)
+{
+}
+
+
+//-------------------------------------------------
+//  mapper specific start/reset
+//-------------------------------------------------
+
+
+void pce_cdsys3_base_device::device_start()
+{
+	/* Set up Arcade Card RAM buffer */
+	m_ram = make_unique_clear<uint8_t[]>(0x30000);
+
+	save_pointer(NAME(m_ram), 0x30000);
```

Comment:
If you’re always instantiating the RAM and it’s always the same size, why not use a `memory_share_creator`?